src/Microsoft.ComponentDetection.Common/IFileWritingService.cs

```diff
@@ -1,8 +1,10 @@
 ﻿namespace Microsoft.ComponentDetection.Common;
+
+using System;
 using System.IO;
 
 // All file paths are relative and will replace occurrences of {timestamp} with the shared file timestamp.
-public interface IFileWritingService
+public interface IFileWritingService : IDisposable
```

Comment:
What about `IAsyncDisposable`? `StreamWriter` implements `IDisposable` and `IAsyncDisposable`